Everest upgrade (#101)
percona/everest-doc: docs/upgrade_with_cli.md

# Upgrade Percona Everest using the CLI

Percona Everest regularly releases updates that contain bug fixes, security improvements, and other enhancements to improve the overall performance of your database.

## Before you upgrade

!!! note

>      Ensure that you have installed the latest version of the CLI.

Reviewer: It is recommended to have the latest version. It's not required though.

Author: Fixed.

> Upgrades to Percona Everest are supported from version 0.9.0 onwards. However, you can only upgrade to one version at a time. For instance, you can upgrade from version 0.9.1 to version 0.10.0, but not directly from version 0.8 to version 0.10.  

Reviewer 2: We should say that users need to download CLI version >=0.10.0 for the upgrade command to work.

## Upgrade

During the upgrade of Percona Everest, only Everest and Everest operator are upgraded, whereas the database operators, database clusters and backups remain unchanged.

!!! note
    It is recommended that you create backups of your databases prior to upgrading Percona Everest.

To upgrade Percona Everest using the CLI, execute the following command:

    everestctl upgrade

??? example "Expected output"

## How to address a failed upgrade

If the upgrade fails, you can attempt it again. If the issue persists, [create a GitHub issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-repository).

## After your upgrade is complete

> After you successfuly upgrade your Percona Everest version, launch the Everest UI. In this example, this is [http://127.0.0.1:8080](http://127.0.0.1:8080).

Reviewer 2: We don't mention anywhere in this page 127.0.0.1 maybe we should refer users back to the last 2 steps of the installation instructions.

Author: Fixed.

# Upgrade Percona Everest using the CLI

Percona Everest regularly releases updates that contain bug fixes, security improvements, and other enhancements to improve the overall performance of your database.

## Before you upgrade

!!! note
     Ensure that you have installed the latest version of the CLI.

Upgrades to Percona Everest are supported from version 0.9.0 onwards. However, you can only upgrade to one version at a time. For instance, you can upgrade from version 0.9.1 to version 0.10.0, but not directly from version 0.8 to version 0.10.  

## Upgrade

> During the upgrade of Percona Everest, only Everest and Everest operators are upgraded, whereas the databases and backups remain unchanged.

Reviewer 2: ```suggestion
During the upgrade of Percona Everest, only Everest and Everest operator are upgraded, whereas the database operators, database clusters and backups remain unchanged.
```

!!! note
    It is recommended that you create backups of your databases prior to upgrading Percona Everest.

To upgrade Percona Everest using the CLI, execute the following command:

    everestctl upgrade

??? example "Expected output"

> To upgrade Percona Everest with the **Debug** mode on, execute the following command:
> 
>     everestctl -v upgrade
> 
> ??? example "Expected output"

Reviewer 2: I don't think we need this extra info.
```suggestion
```

# Upgrade Percona Everest using the CLI

Percona Everest regularly releases updates that contain bug fixes, security improvements, and other enhancements to improve the overall performance of your database.

## Before you upgrade

!!! note
     It is recommended to install the latest version of the CLI .

> You need to download CLI version >=0.10.0 for the upgrade command to work. You can only upgrade to one version at a time. For instance, you can upgrade from version 0.9.1 to version 0.10.0, but not directly from version 0.8 to version 0.10.  

Reviewer 2: ```suggestion
You need to download CLI version >=0.10.0 for the upgrade command to work. You can only upgrade one minor version at a time. For instance, you can upgrade from version 0.9.0 to version 0.10.0, but not directly from version 0.8.0 to version 0.10.0.  
```

## Upgrade

!!! note
    It is recommended that you create backups of your databases prior to upgrading Percona Everest.

During the upgrade of Percona Everest, only Everest and Everest operator are upgraded, whereas the database operators, database clusters and backups remain unchanged.
{.power-number}

1. To upgrade Percona Everest using the CLI, execute the following command:

>     everestctl upgrade

Reviewer 2: Not formatted correctly

Author: Fixed

Reviewer 2: Did you forget to push the commit? I still see the same...

Reviewer 2: it's correct now

> ??? example "Expected output"

Reviewer 2: Missing the example

Author: When the release branch is ready Manish will provide the expected output.

Author: Added the example.

# Upgrade Percona Everest using the CLI

Percona Everest regularly releases updates that contain bug fixes, security improvements, and other enhancements to improve the overall performance of your database.

## Before you upgrade

!!! note
     It is recommended to install the latest version of the CLI .

You need to download CLI version >=0.10.0 for the upgrade command to work. You can only upgrade one minor version at a time. For instance, you can upgrade from version 0.9.0 to version 0.10.0, but not directly from version 0.8.0 to version 0.10.0.  

## Upgrade

!!! note
    It is recommended that you create backups of your databases prior to upgrading Percona Everest.

During the upgrade of Percona Everest, only Everest and Everest operator are upgraded, whereas the database operators, database clusters and backups remain unchanged.
{.power-number}

1. To upgrade Percona Everest using the CLI, execute the following command:

    ```sh
    everestctl upgrade
    ```

    ??? example "Expected output"
        ```

>         ./everestctl upgrade

Reviewer 2: ```suggestion
        everestctl upgrade
```
